pyproject.toml

```
@@ -218,7 +218,7 @@ testpaths = "tests/e3sm_diags"
 python_version = "3.10"
 check_untyped_defs = true
 ignore_missing_imports = true
-warn_unused_ignores = true
+warn_unused_ignores = false
```

```suggestion
warn_unused_ignores = true
```

I think we should keep this because with new releases of `mypy`, old `# type: ignore` comments might no longer be relevant and we can remove them.

---

This flag is flipped for fixing pre-comming issue documented https://github.com/E3SM-Project/e3sm_diags/issues/977.

---

The unused type ignore warnings were resolved in #967 by deleting them. I committed my suggestion to flip it back to `true`. Let's see if it still fails.

---

I reverted addition of `# type: ignore` statements to align with `main` (https://github.com/E3SM-Project/e3sm_diags/pull/974/commits/2d7a20d9be4b7fdfb22010d9d4fb89033884ef3f) and now `pre-commit` passes.